lib/types.js

```diff
@@ -2094,6 +2094,10 @@ RecordType.prototype._createConstructor = function () {
   var innerArgs = [];
   var ds = []; // Defaults.
   var innerBody = '';
+  // `Error.captureStackTrace` is only available in V8 environments
+  if (this._isError && Error.captureStackTrace) {
+    innerBody = 'Error.captureStackTrace(this);\n';
```

Comment:
passing `this.constructor` right? I normally do that forgot here somehow, I'll add it in